app/components/hideable-area.tsx/hideable-area.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { ReactNode, useEffect, useState } from "react"
+import { Text } from "@rneui/themed"
+import { StyleProp, View, ViewStyle } from "react-native"
+
+interface HideableAreaProps {
+  children: ReactNode
+  isHidden?: boolean
+  hideBalance: boolean
+  hiddenContent?: ReactNode
+  styles?: StyleProp<ViewStyle>
+}
+
+const HideableArea: React.FC<HideableAreaProps> = ({
+  children,
+  isHidden,
+  hideBalance,
+  hiddenContent,
+  styles,
+}) => {
+  const [shouldBeHidden, setShouldBeHidden] = useState(isHidden ?? hideBalance)
+
+  useEffect(() => {
+    if (isHidden === undefined) {
+      setShouldBeHidden(hideBalance)
+    } else {
+      setShouldBeHidden(isHidden)
+    }
+  }, [isHidden, hideBalance])
+
+  if (shouldBeHidden) {
+    return <View style={styles}>{hiddenContent || <Text>****</Text>}</View>
+  }
+
+  return <View style={styles}>{children}</View>
```

Comment:
there is imo no point of having a wrapper `<View style={styles}>` when this could be outside the component, ie from from the parent.